File: node-js/src/entities/endpoint.ts
import { Root } from './root';

export interface Endpoint extends Root {
  channels: Array<string>;

  enabled: boolean;

  eventTypes: Array<string>;

  headers: { [key: string]: string };

  name: string;

  secrets: Array<string>;

  uid: string;

  url: string;
}

File: dotnet/Octohooks.net/Domain/Entities/Endpoint.cs
namespace Octohooks.net.Domain.Entities
{
    public class Endpoint : Root
    {
        public string[] Channels { get; set; }

        public bool Enabled { get; set; }

        public string[] EventTypes { get; set; }

        public Dictionary<string, string> Headers { get; set; }

        public string Name { get; set; }

        public string[] Secrets { get; set; }

        public string Uid { get; set; }

        public string Url { get; set; }
    }
}
